### PR TITLE
use srcset in ws fontpages

### LINKED ISSUE #1935: Use srcSet in WS Frontpages
~~Blocked on https://github.com/bbc/simorgh/pull/1831~~

**Is your feature request related to a problem? Please describe.**
We currently load a 660px image for every WS frontpage image, this is not ideally as our images never even reach that size.

**Describe the solution you'd like**
Using/Adapting the `createSrcset` function found [here](https://github.com/bbc/simorgh/blob/87670059908e03184dc6765bcded678152deb012/src/app/containers/Image/helpers/srcSet.js) and used [here](https://github.com/bbc/simorgh/blob/f1afcb471ec55d095a29c21a24ee8c14c5fd189a/src/app/containers/Image/index.jsx#L54), add src set to ws frontpage images.

The image sizes used in srcset may not be perfect, if this is the case this can be optimised in a future issue.

**Testing notes**
[Tester to complete]

Dev insight: Unit tests

### PR DESCRIPTION
**Pull request by ghost**
Resolves #1935

**Overall change:** Add srcset to images in `StoryPromo` container

**Code changes:**

- Use image path to get `locator` and `originCode`
- Pass `srcset` to figure container

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/StoryPromo/index.jsx [hunk 1]
```diff
@@ -11,13 +11,41 @@ import FigureContainer from '../Figure';
 
 import { ServiceContext } from '../../contexts/ServiceContext';
 import deepGet from '../../lib/utilities/deepGet';
+import createSrcset from '../Image/helpers/srcSet';
+
+const pathToArray = path => path.slice(1).split('/');
+
+const getOriginCode = path => {
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: Hey, can you move these two files into a seperate file please, and write input/output assertion tests for them both :)

File: src/app/containers/StoryPromo/index.jsx [hunk 2]
```diff
@@ -11,13 +11,41 @@ import FigureContainer from '../Figure';
...
-  const src = `https://ichef.bbci.co.uk/news/660${imageValues.path}`;
+
+  const { height, width, path } = imageValues;
+
+  const ratio = (height / width) * 100;
+  const originCode = getOriginCode(path);
+  const locator = getLocator(path);
+  const srcset = createSrcset(originCode, locator, width);
+
+  const src = `https://ichef.bbci.co.uk/news/660${path}`;
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: Can you make `660` a variable with the name "Default image width" please :) Bit clearer

**Review by j-pendlebury — APPROVED** [review]
LGTM 👍

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/StoryPromo/helpers/originCode.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import getOriginCode from './originCode';
+
+describe('getOriginCode', () => {
+  it('should return the origin code give a valid path', () => {
+    const path = '/cpsprodpb/FFC1/production/_107437456_177a9008.jpg';
+    const originCode = getOriginCode(path);
+
+    expect(originCode).toEqual('cpsprodpb');
+  });
```
Comment (dr3) [inline src/app/containers/StoryPromo/helpers/originCode.test.js]:
```suggestion
  });
  
```

File: src/app/containers/StoryPromo/helpers/locator.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const getLocator = path => {
```
Comment (dr3) [inline src/app/containers/StoryPromo/helpers/locator.js]: Soz, `helpers/locator.js` and `helpers/originCode.js` dont make much sence IMO in storyPromo, can you put them both in `imageSrcHelpers` folder please, gives it context as to what its for

File: src/app/containers/StoryPromo/imageSrcHelpers/locator.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const getLocator = path => {
+  if (!path) {
+    return null;
+  }
+
+  const items = path.slice(1).split('/');
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/locator.js:6]:
```suggestion
  const items = path.slice(1).split('/');
  
```

File: src/app/containers/StoryPromo/imageSrcHelpers/locator.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  if (items.length === 0) {
+    return null;
+  }
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/locator.js]:
```suggestion
  }

```

File: src/app/containers/StoryPromo/imageSrcHelpers/locator.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  const locator = items.slice(1).join('/');
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/locator.js:7]:
```suggestion
  const locator = items.slice(1).join('/');

```

File: src/app/containers/StoryPromo/imageSrcHelpers/locator.test.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import getLocator from './locator';
+
+describe('getLocator', () => {
+  it('should return the locator given a valid path', () => {
+    const path = '/cpsprodpb/FFC1/production/_107437456_177a9008.jpg';
+    const locator = getLocator(path);
+
+    expect(locator).toEqual('FFC1/production/_107437456_177a9008.jpg');
+  });
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/locator.test.js:9]:
```suggestion
  });

```

File: src/app/containers/StoryPromo/imageSrcHelpers/originCode.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+const getOriginCode = path => {
+  if (!path) {
+    return null;
+  }
+
+  const items = path.slice(1).split('/');
+  const [originCode] = items;
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/originCode.js:7]:
```suggestion
  const [originCode] = items;

```

File: src/app/containers/StoryPromo/imageSrcHelpers/originCode.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  if (!originCode) {
+    return null;
+  }
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/originCode.js:10]:
```suggestion
  }

```

File: src/app/containers/StoryPromo/imageSrcHelpers/originCode.test.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import getOriginCode from './originCode';
+
+describe('getOriginCode', () => {
+  it('should return the origin code give a valid path', () => {
+    const path = '/cpsprodpb/FFC1/production/_107437456_177a9008.jpg';
+    const originCode = getOriginCode(path);
+
+    expect(originCode).toEqual('cpsprodpb');
+  });
```
Comment (dr3) [inline src/app/containers/StoryPromo/imageSrcHelpers/originCode.test.js:9]:
```suggestion
  });

```

**Review by dr3 — CHANGES_REQUESTED** [review]
Spacing is a bit off, can we have new lines after collections of consts or functions please.

Feel free to batch commit these

**Review by dr3 — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]

**Comment by dr3** [comment]
Oops @sadickisaac looks like eslint is complaining 🙈

**Review by dr3 — APPROVED** [review]

**Review by j-pendlebury — APPROVED** [review]
👍 🎉

**Comment by david-boydell** [comment]
Looks good, ready for merge.

**Pull request merged**